Users/danhellem/device samples (#1085)
Azure/azure-iot-sdk-node: device_samples/getting_started/src/send_messages/README.md

# Send messages

> Sending telemetry messages is the first and most basic thing to do using the SDK. There are multiple examples below and each example covers multipe protocols. We suggest you start with the [send messages to IoT Hub](#send-messages-to-hub) to ensure you successfull connection and message send using MQTT. Then you can work work your way through the different protocols.

Reviewer: ```suggestion
Sending telemetry messages is the first and most basic thing to do using the SDK. There are multiple examples below and each example covers multiple protocols. We suggest you start with the [send messages to IoT Hub](#send-messages-to-hub) to ensure you successful connection and message send using MQTT. Then you can work your way through the different protocols.
```

Also note, that all of these examples use the [connection string](../connections/connection_string.js) method to create a connection client. See the [connections](../connections) section if you want use a different connection method.

> # 🦉 Getting setup

Reviewer: ```suggestion
# 🦉 Getting set up
```

| :--------------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------- |
| IOTHUB_DEVICE_DPS_ENDPOINT   | Option and the default is `global.azure-devices-provisioning.net`. You only need to set the variable if you chose to use a different endpoint. |
| IOTHUB_DEVICE_DPS_ID_SCOPE   | `ID scope`                                                                                                                                     |
| IOTHUB_DEVICE_DPS_DEVICE_ID  | `Device ID`                                                                                                                                    |
| IOTHUB_DEVICE_DPS_DEVICE_KEY | `Primary Key` or `Secondary key`                                                                                                               |

> For a more in-depth example of plug and play capabilies, see [this how to sample](../../../how_to_guides/src/plug_and_play)

#### Running the sample

From the `getting_started/src/send_messages` directory, run `node send_messages_to_central.js`

### Send messages in batch

> Batch up several messages and send them in a single HTTP request. |

Reviewer: ```suggestion
Batch up several messages and send them in a single HTTP request.
```
